### PR TITLE
Update SSO guides to take into account new commands: `tctl sso test`, `tctl sso configure`.

### PR DESCRIPTION
**Pull request by Tener**
Planned scope of this PR:

- [x] ~~CLI reference for `tctl sso test`, `tctl sso configure`.~~ Moved to PR: https://github.com/gravitational/teleport/pull/13148
- [x] Update to Github SSO guide
- [x] Updates to Enterprise SSO page
- [ ] Individual IdP guides
  - [x] Okta guide
  - [x] Azure Active Directory (AD) guide
  - [x] Active Directory (ADFS) guide
  - [ ] Google Workspace guide
  - [x] GitLab guide
  - [ ] OneLogin guide
  - [ ] OIDC guide

Parent: https://github.com/gravitational/teleport/issues/9270

### CODE REVIEW

**Comment by Tener** [comment]
If possible, I'd like to get your feedback regarding the changes so far. The is a lot of shared language between individual pages, especially the guides for individual IdPs. Having your feedback at this stage will make it easier to maintain consistency while writing the remaining guides (see the list on top).

File: docs/pages/setup/admin/github-sso.mdx [hunk 1]
```diff
@@ -33,10 +33,17 @@ Instructions for creating a GitHub OAuth app are available here:
 
 [Creating an OAuth App](https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app)
 
-## Step 2/3. Create a GitHub authentication connector
+## Step 2/5. Define the GitHub authentication connector
```
Comment (ptgott) [inline docs/pages/setup/admin/github-sso.mdx:36]:
```suggestion
## Step 2/5. Define a GitHub authentication connector
```
The first paragraph of the section uses "a", and we haven't specified a GitHub connector yet.

File: docs/pages/setup/admin/github-sso.mdx [hunk 2]
```diff
@@ -33,10 +33,17 @@ Instructions for creating a GitHub OAuth app are available here:
...
 
-Define a GitHub authentication connector by creating a file called `github.yaml`
-with the following content:
+Define a GitHub authentication connector by creating a file called `github.yaml`.
```
Comment (ptgott) [inline docs/pages/setup/admin/github-sso.mdx:38]:
```suggestion
Define a GitHub authentication connector by running the following command:
```
I think we can remove the instruction to manually create the `github.yaml` file and condense the first two paragraphs of this section.

File: docs/pages/setup/admin/github-sso.mdx [hunk 3]
```diff
@@ -33,10 +33,17 @@ Instructions for creating a GitHub OAuth app are available here:
...
 [Creating an OAuth App](https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app)
 
-## Step 2/3. Create a GitHub authentication connector
+## Step 2/5. Define the GitHub authentication connector
 
-Define a GitHub authentication connector by creating a file called `github.yaml`
-with the following content:
+Define a GitHub authentication connector by creating a file called `github.yaml`. 
+
+You can do it with the following command:
```
Comment (ptgott) [inline docs/pages/setup/admin/github-sso.mdx:40]:
```suggestion
```

File: docs/pages/setup/admin/github-sso.mdx [hunk 4]
```diff
@@ -62,18 +69,66 @@ spec:
...
-The values of `client_id`, `client_secret`, and `redirect_url` come from the
-GitHub OAuth App you created earlier.
+The values of `client_id`, `client_secret` come from the GitHub OAuth App you created earlier. `redirect_url` is automatically configured by [`tctl sso configure github`](../reference/cli.mdx#tctl-sso-configure-github) based on the settings of Teleport cluster. 
 
-Teleport will request only the `read:org` OAuth scope. Read more about OAuth scopes in GitHub's documentation:
+<Admonition
+  type="tip"
+  title="Teams to logins mapping"
+>
+  The flag `--teams-to-logins` (`-r` for short) can be repeated multiple times with each repetition creating additional mapping entry.
```
Comment (ptgott) [inline docs/pages/setup/admin/github-sso.mdx:78]:
```suggestion
The `--teams-to-logins` flag (`-r` for short) can be repeated multiple times. Each repetition will create an additional entry in the `teams_to_logins` mapping of your GitHub authentication connector.
```

**Review by ptgott — COMMENTED** [review]
I've left a partial review. I need to continue reviewing at the top of the new "Step 3/5. Test the connector" section.

**Comment by Tener** [comment]
@ptgott thank you for the review, I discovered there is another SSO docs PR in flight and which is about to land soon: https://github.com/gravitational/teleport/pull/12407. I'll need to merge my changes with those before they are review-worthy...

File: docs/pages/setup/admin/github-sso.mdx [hunk 5]
```diff
@@ -62,18 +69,66 @@ spec:
         - access
 ```
 
-The values of `client_id`, `client_secret`, and `redirect_url` come from the
-GitHub OAuth App you created earlier.
+The values of `client_id`, `client_secret` come from the GitHub OAuth App you created earlier. `redirect_url` is automatically configured by [`tctl sso configure github`](../reference/cli.mdx#tctl-sso-configure-github) based on the settings of Teleport cluster.
```
[the remark this answers is not in the document]
Comment (Tener) [reply docs/pages/setup/admin/github-sso.mdx:72]: Thank you for the improved phrasing.

I'm not sure if including the link constitutes the breaking of reader focus: we are not forcing them to follow the link, but rather making an option to do so if they are interested. This is similar to inline-link style in [Wikipedia](https://en.wikipedia.org/wiki/Teleport_(open-source_software)#Architecture), for example:

> == Architecture ==
> 
> Teleport is written in [Go](Go (programming language)) programming language, and runs on [UNIX](Unix)-compatible operating systems, including Linux, [macOS](macOS), and several [BSD/OS](BSD) variants. Teleport consists of two [executable](executable)s: <code>tsh</code> (command line client) and <code>teleport</code> (server daemon).

Let me know what you think, though.

**Comment by Tener** [comment]
@ptgott I have rebased this PR on top of another, so this is tagged as "Outdated": https://github.com/gravitational/teleport/pull/12941#discussion_r887875325, but this is definitely still relevant. Hoping to hear your thoughts on this.

**Comment by ptgott** [comment]
@Tener Thanks for working on this! Let me know when all guides have been added, and I'll carve out time to give this PR a complete review.

**Comment by ptgott** [comment]
@Tener Also, would it be feasible to split this into multiple PRs, either one per guide or one per ≈250-500 lines of changes? This will make it a lot easier for me to review. We can also can get completed guides merged while others are waiting to be finished.

**Comment by Tener** [comment]
> @Tener Also, would it be feasible to split this into multiple PRs, either one per guide or one per ≈250-500 lines of changes? This will make it a lot easier for me to review. We can also can get completed guides merged while others are waiting to be finished.

No problem, I can split it. I'm waiting for https://github.com/gravitational/teleport/pull/12407 to be merged before I continue the work on this one.

**Pull request closed without merging**